android/src/main/java/com/marigold/rnsdk/RNMessageStreamModule.kt

```diff
@@ -27,20 +29,47 @@ class RNMessageStreamModule (reactContext: ReactApplicationContext, private val
     @VisibleForTesting
     internal var jsonConverter: JsonConverter = JsonConverter()
 
+    private val eventChannel = Channel<Boolean>()
+    private var defaultInAppNotification = displayInAppNotifications
     init {
         messageStream.setOnInAppNotificationDisplayListener(this)
     }
 
     override fun shouldPresentInAppNotification(message: Message): Boolean {
-        try {
-            val writableMap = jsonConverter.convertJsonToMap(message.toJSON())
-            reactApplicationContext
+        if (defaultInAppNotification) {
+            return true
+        }
+
+        return runBlocking {
+            emitWithTimeout(message)
+        }
+    }
+
+    private suspend fun emitWithTimeout(message: Message): Boolean {
+        return withTimeoutOrNull(5000L) {
+            try {
+                val writableMap = jsonConverter.convertJsonToMap(message.toJSON())
+                val emitter = reactApplicationContext
                     .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
-                    .emit("inappnotification", writableMap)
-        } catch (e: JSONException) {
-            e.printStackTrace()
+                emitter.emit("inappnotification", writableMap)
+                eventChannel.receive()
+            } catch (e: JSONException) {
+                e.printStackTrace()
+                true
+            }
+        } ?: true
+    }
+
+    @ReactMethod
+    fun acknowledgeEvent(shouldHandle: Boolean) {
```

Comment:
`acknowledgeEvent` is a little generic. We should probably rename to something more clear like `inAppHandled(handled: Boolean)` or `notifyInAppHandled(handled: Boolean)` so it's clearer what the action is.